lib/plugins/inspec-plugin-manager-cli/test/functional/inspec-plugin_test.rb

```diff
@@ -77,29 +77,29 @@ class PluginManagerCliList < Minitest::Test
   include PluginManagerHelpers
 
   def test_list_when_no_user_plugins_installed
+    skip_windows!
     result = run_inspec_process_with_this_plugin("plugin list")
 
-    skip_windows!
     assert_includes result.stdout, "0 plugin(s) total", "Empty list should include zero count"
 
     assert_exit_code 0, result
   end
 
   def test_list_all_when_no_user_plugins_installed
-    result = run_inspec_process_with_this_plugin("plugin list --all")
-
     skip_windows!
-    assert_includes result.stdout, "6 plugin(s) total", "--all list should find six"
-    assert_includes result.stdout, "inspec-plugin-manager-cli", "--all list should find inspec-plugin-manager-cli"
-    assert_includes result.stdout, "habitat", "--all list should find habitat"
-
-    assert_exit_code 0, result
-
-    # TODO: split
-    result = run_inspec_process_with_this_plugin("plugin list -a")
-
-    assert_includes result.stdout, "6 plugin(s) total", "-a list should find six"
-
+    result = run_inspec_process_with_this_plugin("plugin list --all")
+    plugin_lines = result.stdout.lines.filter { |line| line.match(/^\s*(inspec|train)-/) }
+
+    # Look for a specific plugin of each type - core, bundle, and system
+    [
+      { name: "inspec-plugin-manager-cli", type: "core" },
+      { name: "inspec-supermarket", type: "bundle" },
+      { name: "train-aws", type: "gem (system)" },
+    ].each do |test_case|
+      plugin_line = plugin_lines.detect { |line| line.include? test_case[:name] }
+      refute_nil plugin_line, "#{test_case[:name]} should be detected in plugin list --all output"
+      assert_includes plugin_line, test_case[:type], "#{test_case[:name]} should be detected as a '#{test_case[:type]}' type in list --all "
```

Comment:
and thus has a bunch of redundancies here

Reply:
Yes, that's right. It's aiming for clarity, not brevity. If one assertion fails, you know exactly what to go fix. 

I believe in tests as useful diagnostic tools; I understand you disagree, but this is still valid.